.github/workflows/ci.yml

```diff
@@ -90,10 +90,16 @@ jobs:
 
     - name: Check Code Quality
       shell: bash -l {0}
+      env:
+         COPYRIGHT: ${{ github.event_name == 'pull_request' && 'PR' || '' }}
       run: |
-        SKIP=flake8,black,black-jupyter pre-commit run --all-files
+        SKIP=flake8,black,update_copyright pre-commit run --all-files
         flake8 --count
         black --check .
+        if [ $COPYRIGHT ]; then
+           git fetch origin $GITHUB_BASE_REF $GITHUB_HEAD_REF
+           pre-commit run update_copyright --from-ref origin/$GITHUB_BASE_REF --to-ref origin/$GITHUB_HEAD_REF
```

Comment:
This is good, but it'll run on every PR but that's unavoidable right?
It's difficult to know when the year changed

Reply:
Exactly and some files might be updated much later in the year than others (if at all)